subsys/logging/dictionary/Kconfig

```diff
@@ -15,3 +15,30 @@ config LOG_DICTIONARY_SUPPORT
 
 	  Normally this is selected by the backend automatically and does
 	  not need to be enabled manually.
+
+config LOG_DICTIONARY_BACKEND_UART
+	bool "Enable UART backend for dictionary based logging"
+	depends on LOG2 && UART_CONSOLE
+	select LOG_DICTIONARY_SUPPORT
+	help
+	  When enabled backend is using UART to output dictionary based logs.
```

Comment:
We also have backends here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/logging/Kconfig.backends

Actually, when I build `samples/subsys/logging/dictionary` for nrf52840dk_nrf52840 board, I get both this dictionary backend uart enabled but also RTT, so where will my logs go ?
```
$ cmake -GNinja -DBOARD=nrf52840dk_nrf52840  -Bbuild_52 samples/subsys/logging/dictionary/
$ grep 'CONFIG_LOG_.*BACKEND' build_52/zephyr/.config
# CONFIG_LOG_BACKEND_UART is not set
CONFIG_LOG_BACKEND_RTT=y
...
CONFIG_LOG_DICTIONARY_BACKEND_UART=y
```

And why is a dedicated `LOG_DICTIONARY_BACKEND_UART` needed, how does it differ from, for example using `LOG_BACKEND_UART=y` with dictionary formatting ?

Reply:
I have rolled the dictionary based UART output into the upper level UART backend, where kconfig is used to choose the output.

As for which backend, the log messages will go to all enabled backends, which is by design.